design-toolkit/components/src/providers/portal.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { UNSAFE_PortalProvider } from '@react-aria/overlays';
+import { useIsSSR } from '@react-aria/ssr';
+import {
+  type PropsWithChildren,
+  type RefObject,
+  useEffect,
+  useState,
+} from 'react';
+
+export function PortalProvider({
```

Comment:
Looks like we have similar versions of this in Dialog and Tooltip, with minor differences. If we're going to add this as an abstraction we need to make sure that it's capable of fulfilling both of those needs